lib/hdb/common.c

```diff
@@ -153,6 +154,23 @@ _hdb_fetch_kvno(krb5_context context, HDB *db, krb5_const_principal principal,
 	    krb5_data_free(&value);
 	    return ret;
 	}
+
+	if ((flags & HDB_F_FOR_AS_REQ) && (flags & HDB_F_CANON) == 0) {
+	    krb5_principal tmp;
```

Comment:
You don't need a `tmp` variable...

Reply:
Wanted to be sure allocation succeeded before swapping in the new principal.